### DISCUSSION TITLE
Unexpected behavior of using downloaded configs between environments : Deploy creates duplicate entries for dashboard and alerting profiles

### DISCUSSION THREAD

**Discussion by avwsolutions** [Dynatrace/dynatrace-configuration-as-code#241] [Archive]
Hi All, 

We (me and @AH-ErikPost )  are busy with a migration from Managed Dynatrace (PRD) to Dynatrace SAAS (DTA). We are using Monaco to download the configuration from our managed tenants and apply it to our SAAS DTA. What we currently see is that the dry-run goes well, but actually the deployments created the objects, but after a re-run the same object is created again (so duplicates). We observed this behaviour for dashboards and alerting profiles.  

When we directly download the project from the target SAAS DTA (deploy) environment this behavior is not there anymore and we can see that only one object is maintained. 

The questions arise now how we can use Monaco as a tool for our pipeline automation (Github actions). 

Our goal is to have multiple versions configurations that can both imported in DTA and PRD without created duplicate objects. So we can have configuration made once and applied multiple times at different environments/tenants. 

please advice in this one. We are using the latest monaco release with CLI 2.0 options enabled.

**Comment by patrickpichler** [comment]
Hi, 

Could you please share the dashboard json and alerting profile jsons which cause this behavior?

**Comment by tatomir146** [comment]
@avwsolutions duplicated objects are usually created if monaco is not able to find existing object with the same name. As @patrickpichler suggested, please share JSONs with us so we can have a look.

**Reply by avwsolutions** [reply]
Hi @tatomir146 we already found that there were indeed issues with the download, we need to update/remove. Still we have an issue with a mobile (custom) application. Creation goes well, but after a redeploy this errors stops monaco to proceed further. It seems something with application-mobile objects, since I also got this behaviour with another JSON payload, nothing fancy. We are using monaco latest (1.5.0) release

Don't know this is a bug ?  Otherwise I will create an issue for this.

```
2021-03-05 10:47:18 ERROR 	Failed to upsert DT object MyObject (HTTP 400)!
    Response was: {"code":400,"message":"Constraints violated!","constraintViolations":[{"path":"applicationType","message":"Payload does include parameters that must not be set!","parameterLocation":"PAYLOAD_BODY","location":null}]}, responsible config: saas_dta/application-mobile/myobject.json
2021-03-05 10:47:18 ERROR Errors during deployment! Check log!
```

application-mobile.yaml
```
config:
- myobject: myobject.json
myobject:
- name: MyObject
```

myobject.json
```
{
 "apdexSettings": {
  "frustratedOnError": true,
  "frustratingThreshold": 12000,
  "toleratedThreshold": 3000
 },
 "applicationType": "CUSTOM_APPLICATION",
 "beaconEndpointType": "CLUSTER_ACTIVE_GATE",
 "costControlUserSessionPercentage": 100,
 "iconType": "DESKTOP",
 "name": "{{.name}}"
}
```

**Reply by avwsolutions** [reply]
@tatomir146  ^ ^

**Reply by patrickpichler** [reply]
In order to better debug this, could you maybe also append the request/response log? (also please make sure to remove any sensitive information from there)

You can activate it by setting the `MONACO_REQUEST_LOG=request.log` and `MONACO_RESPONSE_LOG=response.log` environment variables. ([Here](https://github.com/dynatrace-oss/dynatrace-monitoring-as-code#logging-all-requestsresponse-send-to-dynatrace) a link to the documentation)

**Reply by tatomir146** [reply]
From the error message it looks like `applicationType` parameter is not needed. Can you try removing it?
